dask_expr/io/io.py

```diff
@@ -45,12 +45,19 @@ def _meta(self):
     def _divisions(self):
         return self.operand("divisions")
 
-    @property
+    @functools.cached_property
     def _name(self):
-        return self.operand("_name")
+        return (
+            self.operand("name_prefix") + "-" + _tokenize_deterministic(*self.operands)
+        )
 
     def _layer(self):
-        return dict(self.operand("layer"))
+        dsk = dict(self.operand("layer"))
+        # The name may not actually match the layers name therefore rewrite this
+        # using an alias
+        for part, k in enumerate(self.operand("keys")):
+            dsk[(self._name, part)] = k
+        return dsk
```

Comment:
The intuitive fix for me would've been to overwrite `__dask_keys__` instead to point to the appropriate keys but as I explained above, the implementation of `Expr.__dask_keys__` is hard coded in many places.